nexus/db-queries/src/db/datastore/scim_provider_store.rs

```diff
@@ -7,124 +7,1791 @@
+
+            diesel::delete(dsl::silo_group_membership)
+                .filter(dsl::silo_group_id.eq(to_db_typed_uuid(group_id)))
+                .execute_async(conn)
+                .await?;
+        }
+
+        // Cleanup role assignment records for the deleted group
+        {
+            use nexus_db_schema::schema::role_assignment::dsl;
+
+            diesel::delete(dsl::role_assignment)
+                .filter(
+                    dsl::resource_type
+                        .eq(self.authz_silo.resource_type().to_string()),
+                )
+                .filter(dsl::resource_id.eq(self.authz_silo.resource_id()))
+                .filter(dsl::identity_id.eq(to_db_typed_uuid(group_id)))
+                .filter(dsl::identity_type.eq(IdentityType::SiloGroup))
+                .execute_async(conn)
+                .await?;
+        }
+
+        Ok(true)
     }
 }
 
-#[async_trait::async_trait]
-impl ProviderStore for CrdbScimProviderStore {
+fn convert_to_scim_user(
+    silo_user: SiloUserScim,
+    groups: Option<Vec<UserGroup>>,
+) -> StoredParts<User> {
+    StoredParts {
+        resource: User {
+            id: silo_user.id.to_string(),
+            name: silo_user.user_name,
+            active: silo_user.active,
+            external_id: silo_user.external_id,
+            groups,
+        },
+
+        meta: StoredMeta {
+            created: silo_user.time_created,
+            last_modified: silo_user.time_modified,
+            version: "W/unimplemented".to_string(),
+        },
+    }
+}
+
+fn convert_to_scim_group(
+    silo_group: SiloGroupScim,
+    members: Option<IdOrdMap<GroupMember>>,
+) -> StoredParts<Group> {
+    StoredParts {
+        resource: Group {
+            id: silo_group.id.to_string(),
+            display_name: silo_group.display_name,
+            external_id: silo_group.external_id,
+            members,
+        },
+
+        meta: StoredMeta {
+            created: silo_group.time_created,
+            last_modified: silo_group.time_modified,
+            version: "W/unimplemented".to_string(),
+        },
+    }
+}
+
+impl<'a> ProviderStore for CrdbScimProviderStore<'a> {
     async fn get_user_by_id(
         &self,
-        _user_id: &str,
+        user_id: &str,
     ) -> Result<Option<StoredParts<User>>, ProviderStoreError> {
-        return Err(ProviderStoreError::StoreError(anyhow!(
-            "not implemented!"
-        )));
+        let user_id: SiloUserUuid = match user_id.parse() {
+            Ok(v) => v,
+            Err(_) => {
+                return Err(ProviderStoreError::StoreError(anyhow!(
+                    "user id must be uuid"
+                )));
+            }
+        };
+
+        self.opctx
+            .authorize(
+                authz::Action::ListChildren,
+                &SiloUserList::new(self.authz_silo.clone()),
+            )
+            .await
+            .map_err(external_error_to_provider_error)?;
```

Comment:
This should probably check read on the user, not list children on the silo.

https://gist.github.com/david-crespo/afdb4671a7f72126accf19a7eb8f9319#authorization-semantics